Portal/Controllers/WasteController.cs

```diff
@@ -25,6 +26,31 @@ public IActionResult DuringWhichMonth(int? id)
             return View(model);
         }
 
+        [HttpGet]
+        public IActionResult Index(int? id)
+        {
+            if (id == null)
+                id = 1;
+                //throw new ArgumentNullException("id");
+
+            var model = _wasteService.GetWasteModel(id.Value);
+            return View(model);
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Index(int id, WhatHaveYouDoneWasteModel whatHaveYouDoneWaste) 
+        {
+            if (!ModelState.IsValid)
+            {
+                return View(whatHaveYouDoneWaste);
+            }
```

Comment:
I don't think this additional **ModelState.IsValid** check is required. Based on the validation state of the validation attributes on the model class, the view is automatically rendered with the respective validation errors.